homeassistant/components/aidot/light.py

```diff
@@ -0,0 +1,130 @@
+            for device_coordinator in coordinator.device_coordinators.values()
+        }
+
+        if new_lists - lists_added:
+            async_add_entities(
+                AidotLight(hass, coordinator.device_coordinators[device_id])
+                for device_id in new_lists
+            )
+            lists_added |= new_lists
+        elif lists_added - new_lists:
+            removed_device_ids = lists_added - new_lists
+            lists_added = lists_added - removed_device_ids
+
+    coordinator.async_add_listener(add_entities)
+    add_entities()
+
+
+class AidotLight(CoordinatorEntity[AidotDeviceUpdateCoordinator], LightEntity):
+    """Representation of a Aidot Wi-Fi Light."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self, hass: HomeAssistant, coordinator: AidotDeviceUpdateCoordinator
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator)
+        self._attr_unique_id = coordinator.device_client.info.dev_id
+        self._attr_max_color_temp_kelvin = coordinator.device_client.info.cct_max
+        self._attr_min_color_temp_kelvin = coordinator.device_client.info.cct_min
+
+        model_id = coordinator.device_client.info.model_id
+        manufacturer = model_id.split(".")[0]
+        model = model_id[len(manufacturer) + 1 :]
+        mac = format_mac(coordinator.device_client.info.mac)
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._attr_unique_id)},
+            connections={(CONNECTION_NETWORK_MAC, mac)},
+            manufacturer=manufacturer,
+            model=model,
+            name=coordinator.device_client.info.name,
+            hw_version=coordinator.device_client.info.hw_version,
+        )
+        if coordinator.device_client.info.enable_rgbw:
+            self._attr_color_mode = ColorMode.RGBW
+            self._attr_supported_color_modes = {ColorMode.RGBW, ColorMode.COLOR_TEMP}
+        elif coordinator.device_client.info.enable_cct:
+            self._attr_color_mode = ColorMode.COLOR_TEMP
+            self._attr_supported_color_modes = {ColorMode.COLOR_TEMP}
+        else:
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        self._attr_available = self.coordinator.data.online
+        self._attr_is_on = self.coordinator.data.on
+        self._attr_brightness = self.coordinator.data.dimming
+        self._attr_color_temp_kelvin = self.coordinator.data.cct
+        self._attr_rgbw_color = self.coordinator.data.rgbw
+
+    def _handle_coordinator_update(self) -> None:
+        """Update."""
+        self._attr_available = self.coordinator.data.online
+        self._attr_is_on = self.coordinator.data.on
+        self._attr_brightness = self.coordinator.data.dimming
+        self._attr_color_temp_kelvin = self.coordinator.data.cct
+        self._attr_rgbw_color = self.coordinator.data.rgbw
```

Comment:
extract this in a separate function and call that function at the end of the constructor to deduplicate